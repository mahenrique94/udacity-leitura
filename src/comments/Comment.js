import moment from 'moment'

import { getFormatDate } from 'utils/date'

class Comment {

    constructor({ author, body, deleted, id, parentDeleted, parentId, timestamp, voteScore }) {
        this.author = author
        this.body = body
        this.deleted = deleted
        this.id = id
        this.parentDeleted = parentDeleted
        this.parentId = parentId
        this.timestamp = timestamp
        this.voteScore = voteScore
    }

    get date() {
        return moment(this.timestamp).format(getFormatDate(true))
    }

}

export default Comment
