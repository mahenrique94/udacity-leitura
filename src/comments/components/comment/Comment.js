import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { Media } from 'reactstrap'

import CommentContent from './CommentContent'
import CommentFooter from './CommentFooter'
import CommentHeader from './CommentHeader'

import CommentModel from '@/comments/Comment'

const StyledComment = styled(Media)`
    flex-direction: column;
`

const Comment = ({ comment, edit, remove, vote }) => (

    <StyledComment className="mt-5">
        <Media body>
            <CommentHeader comment={ comment }/>
            <CommentContent comment={ comment }/>
        </Media>
        <CommentFooter comment={ comment } edit={ edit } remove={ remove } vote={ vote }/>
    </StyledComment>

)

Comment.propTypes = {
    comment: PropTypes.objectOf(CommentModel).isRequired,
    edit: PropTypes.func.isRequired,
    remove: PropTypes.func.isRequired,
    vote: PropTypes.func.isRequired
}

export default Comment
