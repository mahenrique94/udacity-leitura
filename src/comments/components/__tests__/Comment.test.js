import React from 'react'
import renderer from 'react-test-renderer'

import Comment from '../comment/Comment'

import CommentModel from '../../Comment'

describe('Tests for Comment component', () => {

    const REQUIRED_PROPS = {
        comment: new CommentModel({
            author: 'Test',
            body: 'Testing...',
            deleted: false,
            id: 'abc123',
            parentDeleted: false,
            parentId: '123abc',
            timestamp: 123123123123,
            voteScore: 1
        }),
        edit: () => console.log('Editing...'),
        remove: () => console.log('Removing...'),
        vote: () => console.log('Voting...')
    }

    it("Should rendering without crash", () => {
        const tree = renderer.create(
            <Comment { ...REQUIRED_PROPS }/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

})
