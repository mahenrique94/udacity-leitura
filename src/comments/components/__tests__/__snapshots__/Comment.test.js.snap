// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Comment component Should rendering without crash 1`] = `
<div
  className="mt-5 sc-iwsKbI kRjAdc media"
>
  <div
    className="media-body"
  >
    <h4
      className="media-heading"
    >
      Test
    </h4>
    Testing...
  </div>
  <div
    className="sc-dnqmqq hMmkfQ sc-htoDjs ldLlmj"
  >
    <div
      className="sc-gzVnrw gQVUgH"
    >
      <button
        className="mr-2 btn btn-success btn-sm"
        onClick={[Function]}
        type="button"
      >
        <span
          className="mr-2"
          style={
            Object {
              "verticalAlign": "middle",
            }
          }
        >
          1
        </span>
      </button>
      <button
        className="mr-2 btn btn-danger btn-sm"
        onClick={[Function]}
        type="button"
      />
    </div>
    <button
      className="ml-2 btn btn-warning btn-sm"
      onClick={[Function]}
      type="button"
    />
    <button
      className="ml-2 btn btn-danger btn-sm"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
